salt/metalk8s/kubeadm/init/certs/apiserver.sls

```diff
@@ -0,0 +1,34 @@
+{%- from "metalk8s/map.jinja" import kube_api with context %}
+
+Create kube api private key:
+  x509.private_key_managed:
+    - name: {{ kube_api.cert.public_key }}
+    - mode: 600
+    - makedirs: True
+
+{#- Add FQDN of the server #}
+{%- do kube_api.cert.subjectAltName.get('DNS').insert(0, grains['fqdn']) %}
+
+{#- TODO: Not use all IPs and some logic to get service IP #}
+{%- do kube_api.cert.subjectAltName.get('IP').append(
+         salt['netaddress.list_cidr_ips'](kube_api.service_ip_range)[1]) %}
+{%- for ip in grains['ipv4'] %}
+  {%- if ip != '127.0.0.1' %}
+    {%- do kube_api.cert.subjectAltName.get('IP').append(ip) %}
+  {%- endif %}
+{%- endfor %}
+
+{#- Convert dictionary to string #}
+{%- do kube_api.cert.update(
+         {'subjectAltName': salt['metalk8s_certs.compute_subject_alt_name'](
+                              kube_api.cert.subjectAltName)}) %}
+Generate kube api certificate:
+  x509.certificate_managed:
```

Comment:
I see nothing about cert validity lifetime. Where is this managed?

Reply:
it's the `day_valid` from the defaults.yaml so 365 days for this one and 3650 days for the CA one